StripeConnect/StripeConnect/Source/Internal/Webview/ConnectComponentWebViewController.swift

```diff
@@ -229,4 +239,19 @@ private extension ConnectComponentWebViewController {
         didFailLoadWithError(error)
         activityIndicator.stopAnimating()
     }
+
+    /// Opens the url in the given payload in an ASWebAuthenticationSession and sends the resulting redirect to
+    func openAuthenticatedWebView(_ payload: OpenAuthenticatedWebViewMessageHandler.Payload) {
+        Task { @MainActor in
+            do {
+                // TODO: MXMOBILE-2491 log `component.authenticated_web.*` analytic
+                let returnUrl = try await authenticatedWebViewManager.present(with: payload.url, in: view.window)
+
+                sendMessage(ReturnedFromAuthenticatedWebViewSender(payload: .init(url: returnUrl, id: payload.id)))
+            } catch {
+                // TODO: MXMOBILE-2491 log `component.authenticated_web.error` analytic
+                debugPrint("Error returning from authenticated web view: \(error)")
```

Comment:
If the ConnectComponentWebViewController has a parent ViewController and parent view then I would say at that point it would indicate an integration error. Would we still want to log this to our analytics?

Reply:
The ASWebAuthenticationSession can also return an error, although I wasn't able to trigger this case myself other than `ASWebAuthenticationSessionError.canceledLogin`. In that case, I don't think it's an integration error.

If the app is backgrounded (`canStart` is false) or it's not in the view hierarchy (window is nil), then I do think we want to log it since it means we're triggering an authenticated webview to open when the user hasn't taken any action to do so.

Reply:
- I renamed `noWindow` to `notInViewHierarchy` for clarity
- I also removed the debug descriptions from these errors since we won't be logging them anywhere (originally I had planned to return them to the web layer, but we ended up not doing that)